scalafix-core/src/main/scala/scalafix/internal/config/ScalaVersion.scala

```diff
@@ -0,0 +1,85 @@
+package scalafix.internal.config
+
+import scala.util.Failure
+import scala.util.Success
+import scala.util.Try
+
+import scala.meta.Dialect
+
+import scalafix.internal.config.ScalaVersion._
+
+sealed trait ScalaVersion {
+  val major: MajorVersion
+  val minor: Option[Int]
+  val patch: Option[Int]
+  val dialect: Dialect = major match {
+    case ScalaVersion.Scala2 => scala.meta.dialects.Scala213
+    case ScalaVersion.Scala3 => scala.meta.dialects.Scala3
+  }
+  def value: String = this match {
+    case Major(major) => s"${major.value}"
+    case Minor(major, minorVersion) => s"${major.value}.${minorVersion}"
+    case Patch(major, minorVersion, patchVersion) =>s"${major.value}.${minorVersion}.$patchVersion"
+  }
+}
+
+object ScalaVersion {
+  val scala2: Major = Major(Scala2)
+  val scala3: Major = Major(Scala3)
+
+  case class Major(major: MajorVersion) extends ScalaVersion{
+    override val minor = None
+    override val patch = None
+  }
+  case class Minor(major: MajorVersion, minorVersion: Int) extends ScalaVersion {
+    override val minor: Some[Int] = Some(minorVersion)
+    override val patch = None
+
+  }
+  case class Patch(major: MajorVersion, minorVersion: Int, patchVersion: Int) extends ScalaVersion {
+    override val minor: Some[Int] = Some(minorVersion)
+    override val patch: Some[Int] = Some(patchVersion)
+  }
+
+    sealed trait MajorVersion {
+      val value: Int
+    }
+    case object Scala2 extends MajorVersion {
+      override val value: Int = 2
+    }
+    case object Scala3 extends MajorVersion {
+      override val value: Int = 3
+    }
+
+   object MajorVersion {
+     def from(int: Long): Try[MajorVersion] =
+       int match {
+         case 2 => Success(Scala2)
+         case 3 => Success(Scala3)
+         case _ => Failure(new Exception(s"Major Scala version can be either 2 or 3 not $int"))
+       }
+   }
+
+  private val longPattern = """\d{1,19}"""
```

Comment:
I am guessing we can stick to 2? also you capture ints here, not longs

Reply:
you're right